src/manifolds/SegreWarped.jl

```diff
@@ -0,0 +1,284 @@
+        \dots,
+        \frac{\sphericalangle(x_d, y_d) (y_d - \langle x_d, y_d \rangle x_d)}{\sin(\sphericalangle(x_d, y_d))}
+    \right),
+````
+where ``c`` is determined by ``\lVert \operatorname{log}_p(q) \rVert_{p} = \operatorname{dist}(p, q)``.
+
+For a proof, see theorem 4.4 in [JacobssonSwijsenVandervekenVannieuwenhoven:2024](@cite).
+"""
+function log(M::MetricManifold{ℝ,Segre{ℝ,V},WarpedMetric{A}}, p, q) where {V,A}
+    q_ = closest_representation(M, p, q)
+    if connected_by_geodesic(M, p, q)
+        v = zeros.(size.(p)) # Initialize
+        log!(M, v, p, q_)
+        return v
+    else
+        return Nothing
+    end
+end
+
+function log!(M::MetricManifold{ℝ,Segre{ℝ,V},WarpedMetric{A}}, v, p, q) where {V,A}
+    m_ = m(M, p, q)
+
+    v[1][1] = q[1][1] * cos(A * m_) - p[1][1]
+
+    for (n, xdot, x, y) in zip(V, v[2:end], p[2:end], q[2:end])
+        a = distance(Sphere(n - 1), x, y)
+        xdot .= (y - dot(x, y) * x) * (q[1][1] / p[1][1]) * sinc(A * m_ / pi) / sinc(a / pi)
+    end
+
+    return 0
+end
+
+@doc raw"""
+    function distance(M::MetricManifold{ℝ, Segre{ℝ,V}, WarpedMetric{A}}, p, q)
+
+Riemannian distance between two points `p` and `q` on the warped Segre manifold.
+
+Assume ``p \doteq (\lambda, x_1, \dots, x_d)``, ``q \doteq (\mu, y_1, \dots, y_d) \in \mathcal{S}_A`` are connected by a geodesic. Let
+````math
+    m = \sqrt{\sphericalangle(x_1, y_1)^2 + \dots + \sphericalangle(x_d, y_d)^2}
+````
+and assume ``(\mu, y_1, \dots, y_d)`` is the representation of ``q`` that minimizes ``m``. Then
+````math
+    \operatorname{dist}_{\mathcal{S}_A}(p, q) = \sqrt{\lambda^2 - 2 \lambda \mu \cos(A m) + \mu^2}.
+````
+"""
+function distance(M::MetricManifold{ℝ,Segre{ℝ,V},WarpedMetric{A}}, p, q) where {V,A}
+    q_ = closest_representation(M, p, q)
+    return sqrt((p[1][1] - q[1][1])^2 + 4 * p[1][1] * q[1][1] * sin(A * m(M, p, q_) / 2)^2)
+    # Equivalent to sqrt(p[1][1]^2 + q[1][1]^2 - 2 * p[1][1] * q[1][1] * cos(A * m)) but more stable for small m
+end
+
+@doc raw"""
+    function riemann_tensor(M::MetricManifold{ℝ, Segre{ℝ,V}, WarpedMetric{A}}, p, u, v)
+
+Riemann tensor of the warped Segre manifold at ``p``.
+
+``\mathcal{S}_A`` is locally a warped product of ``ℝ^{+}`` and ``S^{n_1 - 1} \times \dots \times S^{n_d - 1}``. If ``p \doteq (\lambda, x_1, \dots, x_d) \in \mathcal{S}_A`` and ``u``, ``v``, ``w \in T_{(x_1, \dots, x_d)} (S^{n_1 - 1} \times \dots \times S^{n_d - 1}) \subset T_p \mathcal{S}_A`` then
+````math
+    R_{\mathcal{S}_A}(u, v) w = R_{S^{n_1 - 1} \times \dots \times S^{n_d - 1}}(u, v) w + \lambda^{-2}(\langle u, w \rangle_{p} v - \langle v, w \rangle_{p} u).
+````
+``R_{\mathcal{S}_A}`` is zero in the remaining (orthogonal) directions.
+"""
+function riemann_tensor(
+    M::MetricManifold{ℝ,Segre{ℝ,V},WarpedMetric{A}},
+    p,
+    u,
+    v,
+    w,
+) where {V,A}
+    # Can we avoid the deep-copies here? That looks a bit inefficient
```

Comment:
I think they should not be necessary, none of the functions below does modify them.

Can you maybe commute the tenor just on the components 2,... and add zero anyways upfront? The first term does not seem to use the `[1]` component, the last term (l. 261) can probably be adapted accordingly to just use a different inner and add the first summand accordingly?